pkg/initializer/initializer.go

```diff
@@ -45,24 +45,29 @@ import (
 //   - Try to perform an Etcd data restoration from the latest snapshot.
 //   - No snapshots are available, start etcd as a fresh installation.
 func (e *EtcdInitializer) Initialize(mode validator.Mode, failBelowRevision int64) error {
+	logger := e.Logger.WithField("actor", "initializer")
 	metrics.CurrentClusterSize.With(prometheus.Labels{}).Set(float64(e.Validator.OriginalClusterSize))
 	start := time.Now()
 	isEtcdMemberPresent := false
 	ctx := context.Background()
 	var err error
 
 	//Etcd cluster scale-up case
-	if miscellaneous.IsMultiNode(e.Logger.WithField("actor", "initializer")) {
+	if miscellaneous.IsMultiNode(logger) {
+		clientSet, err := miscellaneous.GetKubernetesClientSetOrError()
+		if err != nil {
+			logger.Fatalf("failed to create clientset, %v", err)
+		}
 		m := member.NewMemberControl(e.Config.EtcdConnectionConfig)
-		isEtcdMemberPresent, err = m.IsMemberInCluster(ctx)
-		if !isEtcdMemberPresent && err == nil {
-			retry.OnError(retry.DefaultBackoff, func(err error) bool {
-				return err != nil
-			}, func() error {
+		isScaleup, err := m.IsClusterScaledUp(ctx, clientSet)
```

Comment:
```go
		isScaleUp, err := m.IsClusterScaledUp(ctx, clientSet)
		if err != nil {
			logger.Warnf("Failed to verify cluster scale-up: %v. Proceeding with bootstrap operation.", err)
		}
		if isScaleup && err == nil {
			err = retry.OnError(retry.DefaultBackoff, errors.AnyError, func() error {
				return m.AddMemberAsLearner(ctx)
			})
			if err != nil {
				logger.Errorf("adding learner member failed: %v", err)
				return err
			}
			// Non-voting member (learner) has been successfully added, so no further restoration or validation is needed.
			return nil
		}
```
@ishan16696 
**Nit:** please check error first

Reply:
WDYT `logger.Warnf("Failed to verify cluster scale-up: %v. Proceeding with bootstrap operation.", err)`

Reply:
I will address it as a separate PR as I also need to add need unit test. I hope it make sense.